packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -2146,6 +2146,9 @@ void main() {
       // Text changed, prompt rect goes away.
       controller.text = '12345';
       await tester.pump();
+      // TODO: I think this test has been silently broken for a long time, because
+      // previously, paints..everything() was accidentally skipping every one if its
+      // first calls, and here, the first call fails the expectation
       verifyAutocorrectionRectVisibility(expectVisible: false);
```

Comment:
@LongCatIsLooong any chance you could help give me some input on this? I see you wrote this test originally, and given the complexity of `EditableText`, I'm having a bit of trouble verifying that this is currently not working as expected. Basically, the issue is that here, by changing the text value on the controller, we're expecting there to be no drawn autocorrect prompt rect, but when you step through the painter calls here, we do see that this is painting a rect with the promptRectColor

Reply:
OK I think I got it, so feel free to give the test a look just to make sure the changes to the test make sense to you (and that they still test what it was testing before), but otherwise, I think this should be good 👍🏻

Reply:
`EditableText` handles programmatically text changes (as in, `controller.text = '12345';`) slightly differently from user-initiated text changes (`await tester.enterText(find.byType(EditableText), '12345');`). Does the test still pass if you change this back to `controller.text = '12345';`?

Reply:
Right so that was the only way I was able to get it to pass—I just tried again and when I run it with `controller.text = '12345';` it fails, saying that the `RenderEditable` painted another rect (`Rect.fromLTRB(0.0, 0.0, 14.0,
14.0)`, which is the same shape of the autocorrect rects that are being drawn).

I did a bunch of stepping through breakpoints to see why it would happen when controlling the text directly vs based on users entering text, but ended up unable to figure it out. Just seemed like, for some reason, there was an extra build/paint happening

Reply:
And by the way, there is reason to think that this may have just been silently incorrect the whole time. This particular part of the test that was failing is using the `everything` method to test the painter, and up until I made these changes, the `everything` method was always skipping the first paint method. 

So it could very easily be the case that `EditableText` has always drawn the rect one more time before hiding it—the previous implementation of `everything` would have just been covering up that extra build/paint